### ISSUE TITLE
"Building from source" seems broken with ninja install, with error inreplace failed expected replacement of "@rpath"

### ISSUE THREAD

**Issue by hehongbo** [Homebrew/brew#6006]
**Please note that we will close your issue without comment if you delete, do not read or do not fill out the issue checklist below and provide ALL the requested information. If you repeatedly fail to use the issue template, we will block you from ever submitting issues to Homebrew again.**

- [x] are reporting a bug others will be able to reproduce and not asking a question. If you're not sure or want to ask a question do so on our Discourse: https://discourse.brew.sh
- [x] ran a `brew` command and reproduced the problem with multiple formulae? If it's a problem with a single, official formula (not cask) please file this issue at Homebrew/homebrew-core: https://github.com/Homebrew/homebrew-core/issues/new/choose. If it's a `brew cask` problem please file this issue at https://github.com/Homebrew/homebrew-cask/issues/new/choose. If it's a tap (e.g. Homebrew/homebrew-php) problem please file this issue at the tap.
- [x] ran `brew update` and can still reproduce the problem?
- [x] ran `brew doctor`, fixed all issues and can still reproduce the problem?
- [x] ran `brew config` and `brew doctor` and included their output with your issue?

```
HeHongbos-Mac:~ hehongbo$ brew config
HOMEBREW_VERSION: 2.1.0
ORIGIN: https://github.com/Homebrew/brew
HEAD: 1337da0f891aebb85accb094d63e401da1b53cba
Last commit: 6 days ago
Core tap ORIGIN: https://github.com/Homebrew/homebrew-core
Core tap HEAD: 893d2b0451b482af8ed944e04577d6e8a590f8f8
Core tap last commit: 3 hours ago
HOMEBREW_PREFIX: /usr/local
HOMEBREW_LOGS: /Users/hehongbo/Library/Logs/Homebrew
CPU: octa-core 64-bit sandybridge
Homebrew Ruby: 2.3.7 => /System/Library/Frameworks/Ruby.framework/Versions/2.3/usr/bin/ruby
Clang: 10.0 build 1001
Git: 2.20.1 => /Library/Developer/CommandLineTools/usr/bin/git
Curl: 7.54.0 => /usr/bin/curl
macOS: 10.14.4-x86_64
CLT: 10.2.0.0.1.1552586384
Xcode: N/A
CLT headers: 10.2.0.0.1.1552586384
HeHongbos-Mac:~ hehongbo$ brew doctor
Your system is ready to brew.

```

# What you were trying to do (and why)

Install some package with `brew install` (on a machine that Homebrew was installed on a non-default path, which needs compiles for some package) and got a failure. 
Later I tried on another machine which Homebrew was in default path `/usr/local` (but require to build from source manually) and it got failure, too. 
So I realized it is not dual to some packages have hardcoded path that points to /usr/local, but actually an issue of brew's building system. I don't know if it's broken by 10.14.4 update.

What I tried:
1. `brew upgrade` on machine A (hackintosh with smc set to macmini, 10.14.4, homebrew was inside non-default path) then **failed**
2. Uninstall and reinstall homebrew on machine A, and reinstall my packages with `brew install`, failed at same package.
3. `brew upgrade` on machine B (Macbook, 10.14.4, non-default path), failed at same package
4. `brew install` my package on machine C (hackintosh with smc set to macpro, 10.14.4, default path), succeed with an existing bottle.
5. install brew on a **fresh new** 10.14.4 virtual machine, in default path. with `brew install` it succeed, and with `brew install --build-from-source` it failed.

note that these package **used to install with no problem** on machine A and B, and they failed when upgrading.

# What happened (include command output)

when install packages recursively, some package failed. I posted the error massage with installation of `atk` below, **but seems every package with `ninja install` inside the building process will trigger this error.**

<details>
  <summary>Command output</summary>
  <pre>
  
```
Updating Homebrew...
==> Downloading https://download.gnome.org/sources/atk/2.32/atk-2.32.0.tar.xz
Already downloaded: /Users/hehongbo/Library/Caches/Homebrew/downloads/0a4a582de7cc57ce536f61894004c06026356f309b6973b6b9bc41f9b7cc5aff--atk-2.32.0.tar.xz
==> meson --prefix=/usr/local/Cellar/atk/2.32.0 ..
==> ninja
==> ninja install
Error: An exception occurred within a child process:
  Utils::InreplaceError: inreplace failed
/usr/local/Cellar/atk/2.32.0/share/gir-1.0/Atk-1.0.gir:
  expected replacement of "@rpath" with "/usr/local/Cellar/atk/2.32.0/lib"
```
  
  </pre>
</details>

# What you expected to happen

Install these packages with no problems.

# Step-by-step reproduction instructions (by running `brew` commands)

run 

`brew install --build-from-source atk` 

on macOS 10.14.4 (18E226)

**Comment by amyspark** [comment]
Can you retry with `brew install --build-from-source --verbose atk`. I'd like to see the exception's stacktrace.

**Comment by amyspark** [comment]
@hehongbo this one has been fixed in https://github.com/Homebrew/homebrew-core/commit/037bbea3a05928b5a6b18fe55f047e3d4c1c3c76. Do you have any other example?

**Comment by hehongbo** [comment]
> @hehongbo this one has been fixed in [Homebrew/homebrew-core@037bbea](https://github.com/Homebrew/homebrew-core/commit/037bbea3a05928b5a6b18fe55f047e3d4c1c3c76). Do you have any other example?

yes, you guys are awesome!
when I head to my laptop and `brew upgrade` all my package and even remove problem package with dependencies and reinstall them, the problem seems goes away. 
I'll let you know if any other failures happens that related to this.
A great thanks!
